fingerprint2.js

```diff
@@ -104,19 +107,48 @@
       keys = this.hasLiedOsKey(keys);
       keys = this.hasLiedBrowserKey(keys);
       keys = this.touchSupportKey(keys);
-      var that = this;
-      this.fontsKey(keys, function(newKeys){
-        var values = [];
-        that.each(newKeys, function(pair) {
-          var value = pair.value;
-          if (typeof pair.value.join !== "undefined") {
-            value = pair.value.join(";");
-          }
-          values.push(value);
-        });
-        var murmur = that.x64hash128(values.join("~~~"), 31);
-        return done(murmur, newKeys);
-      });
+
+      //Asynchronous keys, must be the case sensitive name of the asynchronous function
+      //(this is done to protect context integrity)
+
+      var asyncKeyFunctions = ["fontsKey", "mediaDevicesKey"];
+      this.addAsynchronousKeys(keys, asyncKeyFunctions, done);
+    },
+    addAsynchronousKeys: function(currentKeys, asyncFunctions, done) {
```

Comment:
I think this should be done after #13 is done and should be implemented as a promise chain internally